### PR TITLE
Remove some bugs in DeletePCPPeer and DeletePCPMap

### PR DESCRIPTION
**Pull request by vvsvic**

### CODE REVIEW

**Comment by miniupnp** [comment]
could you give some more details about your fixes ?

**Comment by vvsvic** [comment]
pcpserver.c
1. get_peer_rule_by_index and  get_redirect_rule_by_index return 0 if successful
2. char iaddr2[16] is too small for IPv6 address.

iptpinhole.c
1. you will get error message every minute without removal of broken pinhole

File: miniupnpd/pcpserver.c [hunk 1]
```diff
@@ -1063,7 +1063,7 @@ static void DeletePCPMap(pcp_info_t *pcp_msg_info)
 	/* remove all the mappings for this client */
 	int index;
 	unsigned short eport2, iport2;
-	char iaddr2[16];
+	char iaddr2[INET6_ADDRSTRLEN];
```
Comment (ptatrai) [inline miniupnpd/pcpserver.c]: I don't think this holds string representation of IPv6 address.

Comment (miniupnp) [inline miniupnpd/pcpserver.c]: upnp_get_pinhole_info() ?

Comment (vvsvic) [inline miniupnpd/pcpserver.c]: INET_ADDRSTRLEN = 16
INET6_ADDRSTRLEN = 46

**Pull request merged**